File: 0003 - Longest Substring Without Repeating Characters/solution.c
#define max(a,b) (((a)>(b))?(a):(b))
int lengthOfLongestSubstring(char * s){
    int res = 0, start = 0, m[128];
    memset(m, 0xff, sizeof(m));
    for(int i = 0; s[i]; i++){
        if(m[s[i]] != -1) start = max(start, m[s[i]]+1);
        res = max(res, i-start+1), m[s[i]] = i;
    }
    return res;
}

File: 0003 - Longest Substring Without Repeating Characters/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        res, start, d = 0, 0, {}
        for i, c in enumerate(s):
            if c in d:
                start = max(start, d[c]+1)
            res, d[c] = max(res, i-start+1), i
        return res